### PR TITLE
PWN-4725 - Slider button added

### PR DESCRIPTION
**Pull request by 1sakov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4725

## Description of Work

Added slider button, shimmer is still in progress

## Screenshots (Optional)

Screenshot #1 | Screenshot #2 | Screenshot #3
:------------:|:-------------:|:------------:
Screenshot_20220901_114916 | Screenshot_20220901_114928

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt [hunk 1]
```diff
@@ -73,6 +73,12 @@ class NewOnboardingFragment :
             buttonRestoreWalletOnboarding.setOnClickListener {
                 replaceFragment(CommonRestoreFragment.create())
             }
+
+            binding.sliderButton.setupView(
+                R.string.common_signin_button_text,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt:78]: named args?

File: app/src/main/java/org/p2p/wallet/splash/SplashFragment.kt [hunk 2]
```diff
@@ -25,6 +25,8 @@ class SplashFragment :
     }
 
     override fun navigateToSignIn() {
-        replaceFragment(SignInPinFragment.create())
+        replaceFragment(OnboardingRootFragment.create())
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/splash/SplashFragment.kt:28]: revert?

File: app/src/main/res/layout/fragment_new_onboarding.xml [hunk 3]
```diff
@@ -4,8 +4,8 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:id="@+id/contentView"
     android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:background="@color/lime">
+    android:background="@color/night"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_new_onboarding.xml:7]: bg_night

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        super.onDetachedFromWindow()
+    }
+
+    fun setupView(
+        @StringRes textRes: Int,
+        @DrawableRes iconRes: Int,
+        isLight: Boolean
+    ) {
+        with(binding) {
+            imageViewAction
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt:63]: :(
lonely call

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        initializeTouchListener()
+    }
+
+    override fun onDetachedFromWindow() {
+        binding.shimmerView.stopShimmer()
+        super.onDetachedFromWindow()
+    }
+
+    fun setupView(
+        @StringRes textRes: Int,
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt:58]: actionTextRes

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+        initializeTouchListener()
+    }
+
+    override fun onDetachedFromWindow() {
+        binding.shimmerView.stopShimmer()
+        super.onDetachedFromWindow()
+    }
+
+    fun setupView(
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt:57]: setupSlider?

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    ) {
+        with(binding) {
+            imageViewAction
+            textViewAction.setText(textRes)
+            imageViewAction.setImageResource(iconRes)
+            setLight(isLight)
+        }
+    }
+
+    fun setLight(isLight: Boolean) {
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt:70]: setLightStyle?

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+        with(binding) {
+            containerOval.setOnTouchListener { view, event ->
+                return@setOnTouchListener when (event.action) {
+                    MotionEvent.ACTION_DOWN -> {
+                        setGradientVisible(isVisible = true)
+                        true
+                    }
+                    MotionEvent.ACTION_MOVE -> {
+                        val lastPosition = measuredWidth - view.width - initialPosition
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt:102]: how about extracting all the branches to corresponding methods?

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+                        } else {
+                            // user slided till the end
+                            onSlideCompleteListener?.invoke()
+                        }
+                        true
+                    }
+                    // we assume we are handling all events,
+                    // otherwise above events are not working correctly
+                    else ->
+                        true
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt:133]:
```suggestion
                    else -> { 
                        true
                    }
```

File: ui-kit/src/main/res/layout/widget_slider_button.xml [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    android:background="@drawable/bg_night_rounded_big"
+    tools:parentTag="androidx.constraintlayout.widget.ConstraintLayout">
+
+    <TextView
+        android:id="@+id/textViewAction"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:textAppearance="@style/UiKit.TextAppearance.SemiBold.Text2"
+        android:textColor="@color/cloud"
```
Comment (gslevinkov) [inline ui-kit/src/main/res/layout/widget_slider_button.xml:18]: text_cloud

File: app/src/main/java/org/p2p/wallet/splash/SplashFragment.kt (hunk 2 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/splash/SplashFragment.kt:28]: pls

**Comment by kabramovp2p** [comment]
screenshots ?

**Comment by 1sakov** [comment]
@kabramovp2p @gslevinkov guys, DO NOT REVIEW YET

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt (hunk 8 above)
Comment (1sakov) [reply ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitSliderButton.kt:101]: Didn't get your idea

**Review by kuznetsov — APPROVED** [review]

File: app/src/main/res/layout/fragment_new_onboarding.xml [hunk 11]
```diff
@@ -4,8 +4,8 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:id="@+id/contentView"
     android:layout_width="match_parent"
-    android:background="@color/night"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:background="@color/lime">
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_new_onboarding.xml:8]: bg_lime

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**